pkg/controllers/user/monitoring/cluster_monitoring_enabled_handler.go

```diff
@@ -36,3 +41,32 @@ func (h *clusterMonitoringEnabledHandler) sync(key string, endpoints *k8scorev1.
 	h.cattleClusterController.Enqueue(metav1.NamespaceAll, h.clusterName)
 	return endpoints, nil
 }
+
+// sync will trigger clusterHandler sync loop when node updated, if Cluster Monitoring is enabling.
+func (h *clusterMonitoringEnabledHandler) syncWindowsNode(key string, node *k8scorev1.Node) (runtime.Object, error) {
+	_, monitoringNamespace := monitoring.ClusterMonitoringInfo()
+	cluster, err := h.cattleClusterLister.Get(metav1.NamespaceAll, h.clusterName)
+	if err != nil || cluster.DeletionTimestamp != nil {
+		return node, err
+	}
+
+	// only consider enabling monitoring
+	if !cluster.Spec.EnableClusterMonitoring {
+		return node, nil
+	}
+
+	if !windowNodeLabel.Matches(labels.Set(node.Labels)) {
+		return node, nil
+	}
+
+	endpoint, err := h.agentEndpointsLister.Get(monitoringNamespace, nodeMetricsWindowsEndpointName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, errors.Wrapf(err, "get windows endpoint %s:%s failed", monitoringNamespace, nodeMetricsWindowsEndpointName)
+	}
+	// enqueue to enable windows node exporter
+	if endpoint == nil {
+		h.cattleClusterController.Enqueue(metav1.NamespaceAll, h.clusterName)
```

Comment:
After enqueue, the cluster controller will get all current windows nodes, and update to the app's answers.

Reply:
I mean the cluster will not enqueue when there is an existing endpoint but it needs to be updated.

Reply:
Yes, this only creates when the endpoint is nil, after the endpoint is created, the ControlPlaneEndpointController 
will handle the endpoint update. https://github.com/rancher/rancher/pull/21606/files/555dc178ef0805ad828f483d71c3ea5960c995cb..70cbc2884845b98d58c7284c4dc823a4e5c93c70#diff-bdc6f41467fffd07ab545fada1609d04L34

Reply:
It would be confusing when `ControlPlaneEndpointController` handles windows worker nodes. Can we either move  all windows node logics to this `syncWindowsNode` function, or have a better naming with respect to the `ControlPlaneEndpointController`

Reply:
@gitlawr updated